routesrv/eskipbytes.go

```diff
@@ -11,10 +11,35 @@ import (
 
 	ot "github.com/opentracing/opentracing-go"
 	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/metrics"
 	"github.com/zalando/skipper/routing"
 	"github.com/zalando/skipper/tracing"
 )
 
+type responseWriterInterceptor struct {
+	http.ResponseWriter
+	statusCode   int
+	bytesWritten int
+}
+
+func (w *responseWriterInterceptor) WriteHeader(statusCode int) {
+	w.statusCode = statusCode
+	w.ResponseWriter.WriteHeader(statusCode)
+}
+
+func (w *responseWriterInterceptor) Header() http.Header {
+	return w.ResponseWriter.Header()
+}
+
+func (w *responseWriterInterceptor) Write(p []byte) (int, error) {
```

Comment:
https://pkg.go.dev/net/http#ResponseWriter

// If WriteHeader has not yet been called, Write calls
// WriteHeader(http.StatusOK) before writing the data.

I guess we need to if status is 0 (i.e. WriteHeader was not called) and use 200 instead.

Reply:
this we do now